### ISSUE TITLE
Doesn't find key when using VEYON_AUTH_KEY_NAME

### ISSUE THREAD

**Issue by johannesegger** [veyon/veyon#797]
We're using key-based authentication with different keys for each class room. We don't want anyone spying at computers in other rooms so this seemed reasonable. However as an administrator I sometimes want to spy in other rooms which is why I imported all of the keys.

```
> & "C:\Program Files\Veyon\veyon-cli.exe" authkeys list
teacher_L203_EDV1/private
teacher_L203_EDV1/public
teacher_L204_EDV2/private
teacher_L204_EDV2/public
...
teacher_W204_SLAB/private
teacher_W204_SLAB/public
```

While multiple keys don't work out of the box, the docs say we can use `VEYON_AUTH_KEY_NAME` to use a specific key to change the default behavior of selecting the first accessible private key. Inspecting a single room of my choice is good enough. 

```
> $env:VEYON_AUTH_KEY_NAME
teacher_L203_EDV1
```

However no matter what key name I enter I always get the following error screen after starting Veyon Master:

```
> & "C:\Program Files\Veyon\veyon-master.exe"
```

![image](https://user-images.githubusercontent.com/237707/153595314-764aebaa-1fe0-4207-a58b-27bcba8a7425.png)

Without setting `VEYON_AUTH_KEY_NAME` Veyon seems to pick `teacher_L203_EDV1` because I can access computers using this key. So I don't think the key is damaged.

Any ideas what I'm doing wrong?

EDIT:
After setting the log level to "Debug" the following messages are logged when `VEYON_AUTH_KEY_NAME` is set:

```diff
2022-02-11T14:25:43.317: [DEBUG] PluginManager::loadPlugins(): discovered plugin "AuthKeys" at "C:/Program Files/Veyon/plugins/authkeys.dll"
2022-02-11T14:25:43.317: [DEBUG] PluginManager::loadPlugins(): discovered plugin "BuiltinUltraVncServer" at "C:/Program Files/Veyon/plugins/builtin-ultravnc-server.dll"
2022-02-11T14:25:43.317: [DEBUG] PluginManager::loadPlugins(): discovered plugin "BuiltinDirectory" at "C:/Program Files/Veyon/plugins/builtindirectory.dll"
2022-02-11T14:25:43.317: [DEBUG] PluginManager::loadPlugins(): discovered plugin "Demonstration" at "C:/Program Files/Veyon/plugins/demo.dll"
2022-02-11T14:25:43.317: [DEBUG] PluginManager::loadPlugins(): discovered plugin "DesktopServices" at "C:/Program Files/Veyon/plugins/desktopservices.dll"
2022-02-11T14:25:43.317: [DEBUG] PluginManager::loadPlugins(): discovered plugin "ExternalVncServer" at "C:/Program Files/Veyon/plugins/external-vnc-server.dll"
2022-02-11T14:25:43.332: [DEBUG] PluginManager::loadPlugins(): discovered plugin "FileTransfer" at "C:/Program Files/Veyon/plugins/filetransfer.dll"
2022-02-11T14:25:43.332: [DEBUG] PluginManager::loadPlugins(): discovered plugin "LDAP Basic" at "C:/Program Files/Veyon/plugins/ldap.dll"
2022-02-11T14:25:43.332: [DEBUG] PluginManager::loadPlugins(): discovered plugin "PowerControl" at "C:/Program Files/Veyon/plugins/powercontrol.dll"
2022-02-11T14:25:43.332: [DEBUG] PluginManager::loadPlugins(): discovered plugin "RemoteAccess" at "C:/Program Files/Veyon/plugins/remoteaccess.dll"
2022-02-11T14:25:43.348: [DEBUG] PluginManager::loadPlugins(): discovered plugin "ScreenLock" at "C:/Program Files/Veyon/plugins/screenlock.dll"
2022-02-11T14:25:43.348: [DEBUG] PluginManager::loadPlugins(): discovered plugin "Screenshot" at "C:/Program Files/Veyon/plugins/screenshot.dll"
2022-02-11T14:25:43.348: [DEBUG] PluginManager::loadPlugins(): discovered plugin "SystemUserGroups" at "C:/Program Files/Veyon/plugins/systemusergroups.dll"
2022-02-11T14:25:43.348: [DEBUG] PluginManager::loadPlugins(): discovered plugin "TextMessage" at "C:/Program Files/Veyon/plugins/textmessage.dll"
2022-02-11T14:25:43.348: [DEBUG] PluginManager::loadPlugins(): discovered plugin "UserSessionControl" at "C:/Program Files/Veyon/plugins/usersessioncontrol.dll"
2022-02-11T14:25:43.348: [DEBUG] PluginManager::loadPlugins(): discovered plugin "WebAPI" at "C:/Program Files/Veyon/plugins/webapi.dll"
2022-02-11T14:25:43.379: [DEBUG] VeyonCore::initSystemInfo(): "4.7.1-3-g22ee7d0e" "PC-EGGJ.schule.intern" "winnt" "10.0.19044" "Windows 10 (10.0)" "windows" "10"
+2022-02-11T14:25:43.426: [DEBUG] AuthenticationCredentials::loadPrivateKey(): "C:\\ProgramData\\Veyon\\keys\\private\\teacher_L203_EDV1\\key"
2022-02-11T14:25:50.583: [DEBUG] Logger::~Logger(): Shutdown
```

When `VEYON_AUTH_KEY_NAME` is not set it looks like this:

```diff
2022-02-11T14:26:33.286: [DEBUG] PluginManager::loadPlugins(): discovered plugin "AuthKeys" at "C:/Program Files/Veyon/plugins/authkeys.dll"
2022-02-11T14:26:33.286: [DEBUG] PluginManager::loadPlugins(): discovered plugin "BuiltinUltraVncServer" at "C:/Program Files/Veyon/plugins/builtin-ultravnc-server.dll"
2022-02-11T14:26:33.286: [DEBUG] PluginManager::loadPlugins(): discovered plugin "BuiltinDirectory" at "C:/Program Files/Veyon/plugins/builtindirectory.dll"
2022-02-11T14:26:33.301: [DEBUG] PluginManager::loadPlugins(): discovered plugin "Demonstration" at "C:/Program Files/Veyon/plugins/demo.dll"
2022-02-11T14:26:33.301: [DEBUG] PluginManager::loadPlugins(): discovered plugin "DesktopServices" at "C:/Program Files/Veyon/plugins/desktopservices.dll"
2022-02-11T14:26:33.301: [DEBUG] PluginManager::loadPlugins(): discovered plugin "ExternalVncServer" at "C:/Program Files/Veyon/plugins/external-vnc-server.dll"
2022-02-11T14:26:33.301: [DEBUG] PluginManager::loadPlugins(): discovered plugin "FileTransfer" at "C:/Program Files/Veyon/plugins/filetransfer.dll"
2022-02-11T14:26:33.317: [DEBUG] PluginManager::loadPlugins(): discovered plugin "LDAP Basic" at "C:/Program Files/Veyon/plugins/ldap.dll"
2022-02-11T14:26:33.317: [DEBUG] PluginManager::loadPlugins(): discovered plugin "PowerControl" at "C:/Program Files/Veyon/plugins/powercontrol.dll"
2022-02-11T14:26:33.317: [DEBUG] PluginManager::loadPlugins(): discovered plugin "RemoteAccess" at "C:/Program Files/Veyon/plugins/remoteaccess.dll"
2022-02-11T14:26:33.317: [DEBUG] PluginManager::loadPlugins(): discovered plugin "ScreenLock" at "C:/Program Files/Veyon/plugins/screenlock.dll"
2022-02-11T14:26:33.317: [DEBUG] PluginManager::loadPlugins(): discovered plugin "Screenshot" at "C:/Program Files/Veyon/plugins/screenshot.dll"
2022-02-11T14:26:33.317: [DEBUG] PluginManager::loadPlugins(): discovered plugin "SystemUserGroups" at "C:/Program Files/Veyon/plugins/systemusergroups.dll"
2022-02-11T14:26:33.317: [DEBUG] PluginManager::loadPlugins(): discovered plugin "TextMessage" at "C:/Program Files/Veyon/plugins/textmessage.dll"
2022-02-11T14:26:33.317: [DEBUG] PluginManager::loadPlugins(): discovered plugin "UserSessionControl" at "C:/Program Files/Veyon/plugins/usersessioncontrol.dll"
2022-02-11T14:26:33.332: [DEBUG] PluginManager::loadPlugins(): discovered plugin "WebAPI" at "C:/Program Files/Veyon/plugins/webapi.dll"
2022-02-11T14:26:33.364: [DEBUG] VeyonCore::initSystemInfo(): "4.7.1-3-g22ee7d0e" "PC-EGGJ.schule.intern" "winnt" "10.0.19044" "Windows 10 (10.0)" "windows" "10"
+2022-02-11T14:26:33.410: [DEBUG] AuthenticationCredentials::loadPrivateKey(): "C:\\ProgramData\\Veyon\\keys\\private\\teacher_L203_EDV1\\key"
2022-02-11T14:26:33.520: [DEBUG] ComputerManager::initLocations(): initializing locations for hostname "pc-eggj"
2022-02-11T14:26:33.520: [DEBUG] ComputerManager::initLocations(): initializing locations for hostname "pc-eggj.schule.intern"
2022-02-11T14:26:33.520: [DEBUG] ComputerManager::initLocations(): initializing locations for host address "fe80::ad4f:f33a:e871:49f6"
2022-02-11T14:26:33.520: [DEBUG] ComputerManager::initLocations(): initializing locations for host address "fe80::443:b0c4:9dac:b3b6"
2022-02-11T14:26:33.520: [DEBUG] ComputerManager::initLocations(): initializing locations for host address "fe80::c93b:4939:e800:85d4"
2022-02-11T14:26:33.520: [DEBUG] ComputerManager::initLocations(): initializing locations for host address "fe80::9566:5f48:1b36:25b2"
2022-02-11T14:26:33.520: [DEBUG] ComputerManager::initLocations(): initializing locations for host address "fe80::c14d:825f:1032:d1ff"
...
```

The highlighted lines show that Veyon essentially tries to load the same key.

**Comment by tobydox** [comment]
Thank you very much for your bug report! This is indeed a bug dating back to the first releases of Veyon 4. It will be fixed in 4.7.2. 

As a workaround (or may be even better solution in general), you can also create a separate key pair on your administration computer, export the public key and import it on *all* computers. During the authentication phase, the Veyon Server loads the public key corresponding to the key file name indicated by the counterpart (Veyon Master).

**Comment by johannesegger** [comment]
Thank you for your answer. Creating a separate key is a great idea - I didn't think of it.
